### PR TITLE
fix: [PIE-3428]: pipeline save button disabled when no changes done

### PR DESCRIPTION
**Pull request by PratyushGar**
##### Summary:

Disabled the pipeline save button if no changes or updates have been performed.

##### Jira Links:

[https://harness.atlassian.net/browse/PIE-3428](https://harness.atlassian.net/browse/PIE-3428)

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by PratyushGar** [comment]
retrigger jest

**Review by yc147 — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvas.test.tsx [hunk 1]
```diff
@@ -151,6 +151,23 @@ describe('Pipeline Canvas - new pipeline', () => {
...
+    const props = getProps()
+    const contextValue = getDummyPipelineCanvasContextValue({ isLoading: false })
+    const { getByText } = render(
+      <TestWrapper>
+        <PipelineContext.Provider value={contextValue}>
+          <PipelineCanvas {...props} />
+        </PipelineContext.Provider>
+      </TestWrapper>
+    )
+    act(() => {
```
Comment (yc147) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvas.test.tsx:164]: You can also just check the  `expect(saveButton.hasAttribute('disabled')).toBe(true)`

Comment (yc147) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvas.test.tsx:164]: Or spy on save pipeline api call and check if it is not called.

**Review by YounixIO — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvas.test.tsx (hunk 1 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvas.test.tsx]: Updated the test using spy on the savePipeline call function method.

**Review by yc147 — APPROVED** [review]

**Review by Tamaraius — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**